### ISSUE TITLE
After app rename, build error: Could not find the built application bundle at build/ios/iphonesimulator/Runner.app

### ISSUE THREAD

**Issue by thomas-forte-mcsteen** [flutter/flutter#41793]
## Steps to Reproduce

1. No application code has changed
2. I updated flutter to the newest version `1.9.1+hotfix.4`
3. Now when I run `flutter run` I get an error after successfully building that Runner.app is not found. 

In that directory though I can see the app file was created with a different name: `./build/ios/iphonesimulator/<my apps name>.app/`

I can do a real build `flutter build ios` with no issue, so I just believe flutter is trying to load the wrong file name into the sim.

## Logs

Sorry for only posting the error logs, the full log was too long.
```
[ +336 ms] Could not find the built application bundle at build/ios/iphonesimulator/Runner.app.
[   +1 ms] Error launching application on iPad Pro (9.7-inch).
[   +3 ms] "flutter run" took 26,278ms.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:477:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:490:18)
#3      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#4      _rootRunUnary (dart:async/zone.dart:1132:38)
#5      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#6      _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#7      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#8      Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#9      Future._completeWithValue (dart:async/future_impl.dart:522:5)
#10     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
#11     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
#12     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
#13     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#14     _rootRunUnary (dart:async/zone.dart:1132:38)
#15     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#16     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#17     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#18     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#19     Future._completeWithValue (dart:async/future_impl.dart:522:5)
#20     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
#21     _rootRun (dart:async/zone.dart:1124:13)
#22     _CustomZone.run (dart:async/zone.dart:1021:19)
#23     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#24     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#25     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#26     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#27     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#28     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.9.1+hotfix.4, on Mac OS X 10.14.6 18G103, locale en-US)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/setup/#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, set ANDROID_HOME to that location.
      You may also want to add it to your PATH environment variable.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
[!] Android Studio (not installed)
[✓] VS Code (version 1.38.1)
[✓] Connected device (1 available)

! Doctor found issues in 2 categories.
```

**Comment by jmagman** [comment]
How did you change the name of your app?  Did you edit your Xcode project, or is it just pulling from the pubspec name?

**Comment by thomas-forte-mcsteen** [comment]
I originally named it in the xcode files and manually change the build version and number. How can I have xcode pull the values from the pub spec?

**Comment by jmagman** [comment]
There are `FLUTTER_` build settings that should be generated from the pubspec when you `flutter build` or run your Xcode project, and by default those are used for build versions and numbers in the Info.plist.  The tooling expects the project and workspace are named Runner, so it's not going to work if you change that name.  Note: that's different from the product name `PRODUCT_NAME` that would be shown in the UI.
There's community interest in allow renames of those files though: https://github.com/flutter/flutter/issues/9767.

You might want to regenerate those Xcode project files if you've renamed them manually.
https://github.com/flutter/flutter/wiki/Workarounds-for-common-issues#generated-project-files-outdated

**Comment by thomas-forte-mcsteen** [comment]
I regenerated the ios and android project files. It does not pull the name from my pub spec into the Xcode display name. It is set to the default 'Runner'. If I leave the display name as Runner everything works fine, but as soon as I change the name in the xcode files I get the same error as above.

Is there a way to have the xcode config pull the following into xcode?
Display Name
Bundle Identifier
Target Deployment

It seems that the pub spec version number does get pulled into the xcode project files correctly.

**Comment by jmagman** [comment]
> I regenerated the ios and android project files. It does not pull the name from my pub spec into the Xcode display name. It is set to the default 'Runner'. If I leave the display name as Runner everything works fine, but as soon as I change the name in the xcode files I get the same error as above.

What do you mean by "change the name in the xcode files"?  Can we get reproduction steps?  If you're talking about changing the name of the actual xcworkspace or xcproject, the tools only work if that stays as Runner, and https://github.com/flutter/flutter/issues/9767 is tracking that feature request.

> 
> Is there a way to have the xcode config pull the following into xcode?
> Display Name
> Bundle Identifier
> Target Deployment
> 
> It seems that the pub spec version number does get pulled into the xcode project files correctly.

I just ran `flutter create  --project-name my_app_name test_create_app` and the bundle identifier was set to `com.example.myAppName` and the Info.plist `CFBundleName` is my_app_name.  

CFBundleShortVersionString is FLUTTER_BUILD_NAME and CFBundleVersion is FLUTTER_BUILD_NUMBER in the Info.plist, which is generated from the pubspec and set in Generated.xcconfig.
You should be able to change display name, bundle identifier, and the other properties in this doc and still build:
https://flutter.dev/docs/deployment/ios#review-xcode-project-settings.

I'm surprised I can't find an issue requesting the PRODUCT_NAME to match a title-case version of `--project-name` but maybe I'm just missing that.

**Comment by jmagman** [comment]
> I'm surprised I can't find an issue requesting the PRODUCT_NAME to match a title-case version of `--project-name` but maybe I'm just missing that.

I created https://github.com/flutter/flutter/issues/41824.

**Comment by felixyin** [comment]
Launching lib/main.dart on iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: XMXKKWB4BQ
Running Xcode build...
Xcode build done.                                           16.4s
Could not find the built application bundle at build/ios/iphoneos/Runner.app.
Error launching application on iPhone.

**Comment by felixyin** [comment]
➜  ~ flutter doctor -v
[✓] Flutter (Channel stable, v1.9.1+hotfix.4, on Mac OS X 10.14.6 18G84, locale zh-Hans-CN)
    • Flutter version 1.9.1+hotfix.4 at /Users/fy/tools/flutter
    • Framework revision cc949a8e8b (5 days ago), 2019-09-27 15:04:59 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/fy/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/fy/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.1, Build version 11A1027
    • CocoaPods version 1.8.1

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 40.0.2
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 39.0.5
    • Dart plugin version 192.6262.58

[✓] VS Code (version 1.38.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1

[✓] Connected device (2 available)
    • iPhone            • 55a7454b0af851ec72fb276cc45d3130f53d9817 • ios • iOS 13.1.1
    • iPhone 11 Pro Max • E7DDE7AA-BB9D-4948-894C-D3781A76209C     • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-1 (simulator)

• No issues found!

**Comment by felixyin** [comment]
➜  ~ dart --version
Dart VM version: 2.5.0 (Fri Sep 6 20:10:36 2019 +0200) on "macos_x64"

**Comment by felixyin** [comment]
i resolve it，thank jmagman：https://github.com/flutter/flutter/issues/41793#issuecomment-537681233

**Comment by dhanrajvermaRepo** [comment]
@felixyin, Can you please tell me how you fixed this issue as I am also facing this issue. Please reply as soon as posible.

**Comment by dhanrajvermaRepo** [comment]
@thomas-forte-mcsteen, Have you solved this issue? I am also facing same problem.

**Comment by snaeji** [comment]
@dhanrajvermaRepo i was able to build and release with the following:

1. Delete the ios folder (make sure to back up all your changes there)
2. Run Flutter create .
3. Make any changes you need to ios (except name changes)
4. Run flutter build and create an archive in Xcode
5. Go to the archive and manually change the names there
6. Release the archive through xcode to App store

**Comment by hariom08** [comment]
> > I regenerated the ios and android project files. It does not pull the name from my pub spec into the Xcode display name. It is set to the default 'Runner'. If I leave the display name as Runner everything works fine, but as soon as I change the name in the xcode files I get the same error as above.
> 
> What do you mean by "change the name in the xcode files"? Can we get reproduction steps? If you're talking about changing the name of the actual xcworkspace or xcproject, the tools only work if that stays as Runner, and #9767 is tracking that feature request.
> 
> > Is there a way to have the xcode config pull the following into xcode?
> > Display Name
> > Bundle Identifier
> > Target Deployment
> > It seems that the pub spec version number does get pulled into the xcode project files correctly.
> 
> I just ran `flutter create --project-name my_app_name test_create_app` and the bundle identifier was set to `com.example.myAppName` and the Info.plist `CFBundleName` is my_app_name.
> 
> CFBundleShortVersionString is FLUTTER_BUILD_NAME and CFBundleVersion is FLUTTER_BUILD_NUMBER in the Info.plist, which is generated from the pubspec and set in Generated.xcconfig.
> You should be able to change display name, bundle identifier, and the other properties in this doc and still build:
> https://flutter.dev/docs/deployment/ios#review-xcode-project-settings.
> 
> I'm surprised I can't find an issue requesting the PRODUCT_NAME to match a title-case version of `--project-name` but maybe I'm just missing that.

@jmagman , I'm also having an issue related to this one. And I know what @thomas-forte-mcsteen is talking about. He says, Manually changing the Display name form 'Runner' to custom name breaks the build while trying to run from flutter, but the build works without any problem while running from xcode directly.
So, my problem is similar to this one. But this time, if i change the 'bundle identifier' instead of 'Display name' in xcode manually, the app crashes from flutter only. But as expected the app builds smoothly from xcode directly. 

My need to change the bundle identifier manually in xcode is to test iOS in-app functionality. For this to work, the bundle identifier should match the bundle ID in the  iOS app store, so that i can test the in-app implementation from the flutter environment. If there is a way to set the bundle identifier

**Comment by jmagman** [comment]
I'm about to put out a pull request with to change the default iOS project template:
- Build setting PRODUCT_NAME will default to the pubspec.yaml name. The instructions at https://flutter.dev/docs/deployment/ios#review-xcode-project-settings Xcode 11 General > Display Name change this build setting.  
- Info.plist CFBundleName will be `$(PRODUCT_NAME)`.  This is the default in a new Xcode project, and the Flutter macOS Info.plist template.  Previously, the iOS Info.plist CFBundleName defaulted to the pubspec name.
- Your built .app bundle will be named whatever Xcode reports to be the `FULL_PRODUCT_NAME`
```
cd ios
/usr/bin/xcodebuild -project Runner.xcodeproj -scheme Runner -showBuildSettings 
...
    FULL_PRODUCT_NAME = My App.app
...
```
- `flutter run` will install `build/ios/iphoneos/<FULL_PRODUCT_NAME>.app.`

**Comment by dpedrinha** [comment]
So if I need different names or versions for the Android and iOS app, I can't. Also, it seams like I can't have localized app names for iOS since it pulls it from pubspeck.yaml. Is that right?

**Comment by jmagman** [comment]
> So if I need different names or versions for the Android and iOS app, I can't. 

@dpedrinha You can, both now and in the PR I put out.  It defaults to the name in the pubspec.yaml, but you can change it to whatever you want.  You can add a [InfoPlist.strings](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/AboutInformationPropertyListFiles.html#//apple_ref/doc/uid/TP40009254-102276) to your project to localize your display name.

**Comment by dpedrinha** [comment]
Infoplist.strings did the trick. Thanks.

**Comment by jmagman** [comment]
Reopening again, the PR caused internal test failures.

**Comment by 0xcodezero** [comment]
Let's differentiate between two options, 

1- **Display Name**: (Open Xcode > Select Runner Project > Select Target "Runner" > On "General" Tab > Identity > Display Name ) Keep it to Runner.
It affects the main runnable file (or Package name), which is <NAME>.app file.

Flutter expects this field to retain it as "Runner". And, this Field will not be displayed to the user. It will affect the name of the Archived file that to be uploaded via the Organizer or Fastlane. and It is related to the "Packaging" section inside Target's Build Settings.

2- **Bundle Name**, or CFBundleName: Defined within **info.plist** 
It defines the name displayed for users and you can change it without any build errors.

**Comment by mirkopoloni** [comment]
> Let's differentiate between two options,
> 
> 1- **Display Name**: (Open Xcode > Select Runner Project > Select Target "Runner" > On "General" Tab > Identity > Display Name ) Keep it to Runner.
> It affects the main runnable file (or Package name), which is .app file.
> 
> Flutter expects this field to retain it as "Runner". And, this Field will not be displayed to the user. It will affect the name of the Archived file that to be uploaded via the Organizer or Fastlane. and It is related to the "Packaging" section inside Target's Build Settings.
> 
> 2- **Bundle Name**, or CFBundleName: Defined within **info.plist**
> It defines the name displayed for users and you can change it without any build errors.

@jmagman if this is true then maybe the doc should be updated here https://flutter.dev/docs/deployment/ios#review-xcode-project-settings
I also add the same problem because it was the first time that I had to release an app and in the doc it says: 

> In the Identity section:
> 
> Display Name
> The name of the app to be displayed on the home screen and elsewhere.

> 

So I wrongly changed the name, an update could prevent new users from facing the same problem.

**Comment by sanekyy** [comment]
Here good solution:

https://github.com/flutter/flutter/issues/25370#issuecomment-540947710

**Comment by MATTYGILO** [comment]
I ran flutter clean and then it now it no longer works when i run the code.

**Comment by eduardonunesp** [comment]
Same here even with a project created from scratch 

```bash
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale en-BR)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.42.1)
[✓] Connected device (1 available)

• No issues found!flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.12.13+hotfix.8, on Mac OS X 10.15.3 19D76, locale en-BR)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.42.1)
[✓] Connected device (1 available)

• No issues found!
```

```bash
flutter create --project-name myapp myapp
```

Updated the name on xcode `Display Name`

```bash
cd myapp
open ios/Runner.xcodeproj
```
Before

![image](https://user-images.githubusercontent.com/582516/74935437-007cd600-53c7-11ea-96af-367a5edeb6cb.png)

After 

![image](https://user-images.githubusercontent.com/582516/74935468-0f638880-53c7-11ea-8733-04316459a7c0.png)

```bash
flutter clean && flutter run
```

```bash
Cleaning Xcode workspace...                                         2.4s
Deleting build...                                                    2ms
Deleting .dart_tool...                                               0ms
Launching lib/main.dart on iPhone 11 Pro in debug mode...
Running Xcode build...

 ├─Assembling Flutter resources...                           6.8s
 └─Compiling, linking and signing...                         3.9s
Xcode build done.                                           12.8s
Could not find the built application bundle at build/ios/iphonesimulator/Runner.app.
Error launching application on iPhone 11 Pro.
```

```bash
ls build/ios/iphonesimulator/
MyAPP.app
```

**Comment by sarbogast** [comment]
I had the same error and changing the BundleName in Info.plist did the trick. The whole PRODUCT_NAME nonsense didn't work at all or I didn't get how to do it properly.

**Comment by aarajput** [comment]
I was facing same issue, this is how i fixed this issues. 
1- Open Xcode.
2- Change **Display Name** to _Runner_
3- Go to android studio and run your application on your iOS device.
4- After it successfully ran, go back to Xcode and change **Display Name** back to the name of your application.
5- Go to android studio and run you app on your iOS device again. (No error this time).

**Comment by aarajput** [comment]
I was facing same issue, this is how i fixed this issues. 
1- Open Xcode.
2- Change **Display Name** to _Runner_
3- Go to your info.plist file and change **CFBundleName** to your application's name.
4- Now, run your flutter application on your iOS device. (No error this time).

**Comment by xster** [comment]
This is the same issue as https://github.com/flutter/flutter/issues/9767 right?

**Comment by jmagman** [comment]
@xster Not quite, see https://github.com/flutter/flutter/issues/9767#issuecomment-564819538.
I had a fix for this one but it broke the roll because of how logging is parsed on older iOS devices. I need to fix and reland.

**Comment by puggie03** [comment]
I changed the Bundle name in Info.plist and it worked for me

**Comment by woprandi** [comment]
The doc should be updated if the Display Name must not be changed.

**Comment by woprandi** [comment]
The doc has been updated, I think this issue can be closed

**Comment by jmagman** [comment]
This should not be closed, the tool should handle app bundle renames.  https://github.com/flutter/flutter/pull/47266 needs to be relanded with fixes for syslog-style logging.

**Comment by UseItApps** [comment]
I also get this error after running `flutter clean`.

**Comment by UseItApps** [comment]
This is also due to editing the `Display Name` in Xcode, reverted back to `Runner` in Xcode but build is still failing. Re-ran `flutter clean && flutter build` but it is still failing.

**Comment by SamerDiab** [comment]
@UseItApps you answer is right on the money, thanks!

**Comment by Harshalshree** [comment]
@UseItApps @jmagman So what is the final solution to the problem? Please explain step by step.

**Comment by pdliuw** [comment]
I ran into the same problem, I solved the problem through the following steps, hope this helps you

* 1. Open the file: ios-> Runner.xcodeproj-> project.pbxproj


* 2. Modify the parameters: found in the "project.pbxproj" file: "Profile", "Release", "Debug"
Within the "PRODUCT_NAME" attribute and change its value to "PRODUCT_NAME =" $ (TARGET_NAME) ";"

`` `
PRODUCT_NAME = "$ (TARGET_NAME)";
`` `

**Comment by vifird** [comment]
***For Debug***
Open `ios/Runner.xcodeproj/project.pbxproj`, find `/* Debug */ = ` section, then change the value of `PRODUCT_NAME` to `"Runner"` or `"$(TARGET_NAME)"`
***For Release***
Use `ios-deploy` instead of `flutter install`, such as :
```ios-deploy --id ${deviceId} --bundle /xxx/xxx.app```

This is the only solution that i found, after desperated for flutter.

**Comment by BokAugust** [comment]
> I was facing same issue, this is how i fixed this issues.
> 1- Open Xcode.
> 2- Change **Display Name** to _Runner_
> 3- Go to your info.plist file and change **CFBundleName** to your application's name.
> 4- Now, run your flutter application on your iOS device. (No error this time).

Thanks so much!

**Comment by majdkz97** [comment]
> > I was facing same issue, this is how i fixed this issues.
> > 1- Open Xcode.
> > 2- Change **Display Name** to _Runner_
> > 3- Go to your info.plist file and change **CFBundleName** to your application's name.
> > 4- Now, run your flutter application on your iOS device. (No error this time).
> 
> Thanks so much!

That's Work, Thanks.

**Comment by LCostaN** [comment]
Can't we create a field 'display: Display Name' to pubspec.yaml? That way, I don't know if a similar problem might arise, but the idea is to keep things simple.

A display name should be just a display. Change the project name in pubspec for example makes the package handler fail so you can't freely handle that.

**Comment by jmagman** [comment]
This should be fixed in 1.18.0-10.0.pre or later.
